app/src/main/java/com/scrumbums/donationboi/controllers/ListViewCustom.java

```diff
@@ -0,0 +1,206 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import android.widget.ListView;
+import android.app.Activity;
+
+public class ListViewCustom extends AppCompatActivity {
+    private List<ListElement> locationSamples = new ArrayList<>();
+    private ListView listView;
+//    private TextView csvElementView;
+
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.csv_view);
+        readLocationData();
+
+//        csvElementView = findViewById(R.id.csv_text);
+
+        ArrayAdapter adapter = new ArrayAdapter<>(this, R.layout.csv_element_view, locationSamples);
+        listView = findViewById(R.id.mobile_list);
+
+        listView.setAdapter(adapter);
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?>adapter, View v, int position, long id) {
+                ListElement e = locationSamples.get(position);
+                Intent intent = new Intent(ListViewCustom.this, ListElementPage.class);
+                intent.putExtra("Name", e.getName());
+                intent.putExtra("Address", e.getStreetAddress());
+                intent.putExtra("Phone Number", e.getPhoneNumber());
+                intent.putExtra("Longitude", e.getLongitude());
+                intent.putExtra("Latitude", e.getLatitude());
+                intent.putExtra("Type", e.getLocationType());
+
+                Bundle bund = intent.getExtras();
+                startActivity(intent);
+
+
+            }
+
+        });
+    }
+
+    private void readLocationData() {
+
+        InputStream is = getResources().openRawResource(R.raw.location_data);
+        BufferedReader br = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
+
+        String line = "";
+        try {
+            br.readLine();
+            while ((line = br.readLine()) != null) {
+//                Log.d(MainActivity.TAG, line);
+                String[] tokens = line.split(",");
+                ListElement listElement = new ListElement();
+                listElement.setName(tokens[1]);
+                listElement.setLatitude(Float.parseFloat(tokens[2]));
+                listElement.setLongitude(Float.parseFloat(tokens[3]));
+                listElement.setStreetAddress(tokens[4]);
+                listElement.setCity(tokens[5]);
+                listElement.setState(tokens[6]);
+                listElement.setZipCode(Integer.parseInt(tokens[7]));
+                listElement.setLocationType(tokens[8]);
+                listElement.setPhoneNumber(tokens[9]);
+                listElement.setWebsite(tokens[10]);
+                locationSamples.add(listElement);
+
+                Log.d("MainActivity", "Just Created: " + listElement);
+
+            }
+        } catch (IOException e) {
+            Log.e("MainActivity", "error reading assets", e);
```

Comment:
```suggestion
            Log.e(TAG, "error reading assets", e);
```
Change the first variable of Log to `TAG` and then make that a string constant like "DONATION-BOI"